### PR TITLE
feat: implement dealMachine definition + 77-case state × event matrix test (ENG-44)

### PR DESCRIPTION
**Pull request by Connorbelez**
Sequential compound state machine with three nested phases (lawyerOnboarding,
documentReview, fundsTransfer), structural phase gates via type: "final" + onDone,
and DEAL_CANCELLED from all non-terminal states. 99 tests, zero gaps.

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

## Summary by CodeRabbit

* **New Features**
  * Implemented a multi-phase deal closing workflow with states for lawyer onboarding, document review, and funds transfer.
  * Added event-driven transitions for key deal actions (lawyer verification, document signing, fund receipt, cancellation).

* **Tests**
  * Added comprehensive test coverage for deal workflow state machine, including 77 state-event pair scenarios and edge cases.

### CODE REVIEW

File: convex/engine/machines/deal.machine.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		createDealAccess: () => {},
+		archiveSignedDocuments: () => {},
+		confirmFundsReceipt: () => {},
+		commitReservation: () => {},
+		prorateAccrualBetweenOwners: () => {},
+		updatePaymentSchedule: () => {},
+		voidReservation: () => {},
+		notifyCancellation: () => {},
+		revokeAllDealAccess: () => {},
+	},
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/engine/machines/deal.machine.ts]: By design — not a bug. ENG-44 scope is the machine definition + exhaustive test matrix only. The effect-marker actions are intentional no-op stubs (see the machine comment: "Effects are declared as string names, resolved at runtime by the Effect Registry"). The effect implementations will be registered in the effects registry as each is built in later tickets. `scheduleEffects()` already handles unregistered effects gracefully (warns and skips), so no silent data loss occurs — transitions persist correctly, effects are simply not yet wired.
